### DISCUSSION TITLE
Different handling of Features in Profile and InstallableUnit locations

### DISCUSSION THREAD

**Discussion by vaclavHala** [eclipse-tycho/tycho#1210] [Q&A]
Hello, we have two different target files - one for release testing using a materialized product and other for development using `InstallableUnit` http repository. Content of the two targets should be identical, the difference is only where the bundles come from.
In our test bundle we have an extraRequirement like this
```
<plugin>
    <groupId>org.eclipse.tycho</groupId>
    <artifactId>target-platform-configuration</artifactId>
    <configuration>
        <dependency-resolution>
            <extraRequirements>
                <requirement>
                    <type>eclipse-feature</type>
                    <id>com.my.feature</id>
                    <versionRange>0.0.0</versionRange>
                </requirement>
```
With the `InstallableUnit` target (where it is listed as `com.my.feature.feature.group`) this works fine. However with the `Profile` target where it is supposed to come from the installed product it fails to resolve, saying that
```
You requested to install 'org.eclipse.equinox.p2.iu; com.my.feature.feature.group 0.0.0' but it could not be found
```
I tried to debug the problem and look into `TargetDefinitionResolver` to see if it is looking for the feature in the correct place. 
Indeed the feature is found but is presented as `com.my.feature.feature.jar` (classifier `org.eclipse.update.feature`, id `com.my.feature` in `ArtifactKey`) which does not satisfy the requirement. 

In eclipse when I look at the target file it resolves fine and my feature is listed among features, but there may be some abstraction there which treats both `.feature.jar` and `.feature.group` the same which tycho does not do.

So the question is - is the different handling of features by different locations intentional? If so, is there some way I could write the `extraRequirement` such that it is satisfied both when using `InstallableUnit` and `Profile` locations?

**Comment by laeubi** [comment]
> However with the `Profile` target where it is supposed to come from the installed product it fails to resolve, saying that

Profile locations are only rudimentary supported at Tycho, can you provide an [integration-test](https://github.com/eclipse/tycho/wiki#providing-an-integration-test) to demonstrate the issue so we can improve the `Profile` location?

**Reply by vaclavHala** [reply]
In the end we went around the problem by assembling the target different ways in different situations by maven profiles.
I will get back to this if we encounter this problem again in the future
